### PR TITLE
Make back button in onboarding optional

### PR DESCRIPTION
**Pull request by Bettelstab**
Solves #112: Going back to Safe is useless as Safe will redirect again to Onboarding.
Will need another PR in Safe which uses this option.

### CODE REVIEW

File: src/lib/RequestTypes.ts [hunk 1]
```diff
@@ -328,10 +338,11 @@ export class AccountsRequest {
                     ),
                 } as ParsedCheckoutRequest;
             case RequestType.ONBOARD:
-                request = request as SimpleRequest;
+                request = request as OnboardingRequest;
                 return {
                     kind: RequestType.ONBOARD,
                     appName: request.appName,
+                    hideBack: request.hideBack === true,
```
Comment (sisou) [inline src/lib/RequestTypes.ts]: Another possibility (not saying it's better, just possible :wink:)
```suggestion
                    hideBack: !!request.hideBack,
```

File: src/lib/RequestTypes.ts [hunk 2]
```diff
@@ -19,14 +19,20 @@ export enum RequestType {
...
 }
 
 export interface SimpleResult {
     success: true;
 }
 
+export interface OnboardingRequest {
+    kind?: RequestType.ONBOARD;
+    appName: string;
+    hideBack?: boolean;
```
Comment (sisou) [inline src/lib/RequestTypes.ts]: I propose the name `disableBack` for the parameter.

File: src/lib/RequestTypes.ts (hunk 1 above)
Comment (Bettelstab) [reply src/lib/RequestTypes.ts]: Never liked that. No no no :D

Comment (Bettelstab) [reply src/lib/RequestTypes.ts]: Especially, in this case I want that value to be exactly true, and not "nipir32pnp132r" or whatever would be possible with `!!` ;)

**Pull request merged**